### PR TITLE
doc: How to write kcl.mod.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

#8 
#9 

#### 2. What is the scope of this PR (e.g. component or file name):

Design document.

### CODE REVIEW

File: docs/kcl_mod.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# kcl.mod
```
Comment (Peefy) [inline docs/kcl_mod.md:1]: + I don't have much problem with the structure definition. If you want to use it as a specification and user document, you can refer to some better documents for polishing, such as `cargo.toml`, and pay attention to compatibility and other issues. Change `kpm.lock` to `kcl.mod.lock`?
+ `kcl.mod` can use general structure definition methods such as protobuf on the basis of specifications to facilitate multilingual access.

File: docs/kcl_mod.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+## Background
+kpm needs a configuration file for KCL project, which is responsible for marking the root directory, determining dependency versions, configuring compilation options or others.
```
Comment (Peefy) [inline docs/kcl_mod.md:4]: You can use https://github.com/KusionStack/KCLVM/issues/339 and https://github.com/KusionStack/KCLVM/issues/338 in `Background`.

File: docs/kcl_mod.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Capacity
```
Comment (Peefy) [inline docs/kcl_mod.md:6]: Specification or Definition

File: docs/kcl_mod.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+- name: the name of the current project.
```
Comment (Peefy) [inline docs/kcl_mod.md:8]: Lack of structured specification definition of fields, including type, value range, etc.

File: docs/kcl_mod.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+```toml
+[dependencies]
+xxxx = {git = "<github url>", version = "0.4.5"} 
+xxxx = {git = "<github url>", commit = "h87h8f78"} 
+xxxx = {git = "<github url>", branch = "main"} 
+```
+
+## Workspaces
+```toml
+[workspace]
```
Comment (Peefy) [inline docs/kcl_mod.md:50]: What does `workspace` mean in KCL?

File: docs/kcl_mod.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+members = [
+    # subdirectory relative to the current directory, 
+    # which must contains kcl.mod.
+    "xxx" 
+]
+```
+
+## kpm.lock
```
Comment (Peefy) [inline docs/kcl_mod.md:58]: + Just use `kcl.mod.lock`
+ Missing specification definition of lock file.

File: docs/kcl_mod.md (hunk 2 above)
Comment (zong-zhe) [reply docs/kcl_mod.md]: I removed the backgroud in this doc, backgroud should probably be in the README on the front page. This is a kcl.mod spec.

File: docs/kcl_mod.md (hunk 3 above)
Comment (zong-zhe) [reply docs/kcl_mod.md]: Fixed.

File: docs/kcl_mod.md (hunk 6 above)
Comment (zong-zhe) [reply docs/kcl_mod.md]: fixed.

File: docs/kcl_mod.md (hunk 5 above)
Comment (zong-zhe) [reply docs/kcl_mod.md]: removed.

File: docs/kcl_mod.md (hunk 1 above)
Comment (zong-zhe) [reply docs/kcl_mod.md:1]: fixed.

File: docs/kcl_mod.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+```toml
+[dependencies]
+# git release
+xxxx = {git = "<github url>", version = "0.4.5"} 
+# git commit
+xxxx = {git = "<github url>", commit = "h87h8f78"} 
+# git branch
+xxxx = {git = "<github url>", branch = "main"} 
+```
+
```
Comment (zong-zhe) [inline docs/kcl_mod.md:45]: From URL / docker res ?

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**